Python/Product/Debugger/PtvsdVersionHelper.cs

```diff
@@ -51,26 +51,79 @@ internal class PtvsdVersionHelper {
             if (PackageVersion.TryParse(response.Debugger.Version, out PackageVersion runningVersion)) {
                 var bundledPtvsdVersion = PackageVersion.Parse(PtvsdVersion.Version);
                 if (runningVersion.CompareTo(bundledPtvsdVersion) < 0) {
-                    ShowPtvsdVersionMessage(Strings.InstalledPtvsdOutdatedMessage.FormatUI(response.Debugger.Version, PtvsdVersion.Version));
+                    ShowPtvsdMessage(
+                        Strings.InstalledPtvsdOutdatedTitle,
+                        Strings.InstalledPtvsdOutdatedMessage.FormatUI(response.Debugger.Version, PtvsdVersion.Version),
+                        allowDisable: false,
+                        isError: false
+                    );
                 }
             }
         }
 
         public static void VerifyPtvsdVersionError(PtvsdVersionArguments args, ProtocolException ex) {
-            ShowPtvsdVersionMessage(Strings.InstalledPtvsdOutdatedMessage.FormatUI("unknown", PtvsdVersion.Version));
+            ShowPtvsdMessage(
+                Strings.InstalledPtvsdOutdatedTitle,
+                Strings.InstalledPtvsdOutdatedMessage.FormatUI("unknown", PtvsdVersion.Version),
+                allowDisable: false,
+                isError: false
+            );
         }
 
         public static void VerifyPtvsdVersionLegacy() {
-            ShowPtvsdVersionMessage(Strings.InstalledPtvsdOutdatedMessage.FormatUI("3.*", PtvsdVersion.Version));
+            ShowPtvsdMessage(
+                Strings.InstalledPtvsdOutdatedTitle,
+                Strings.InstalledPtvsdOutdatedMessage.FormatUI("3.*", PtvsdVersion.Version),
+                allowDisable: false,
+                isError: false
+            );
         }
 
-        private static void ShowPtvsdVersionMessage(string message) {
-            MessageBox.Show(
-                new Win32Window(Process.GetCurrentProcess().MainWindowHandle),
-                message,
-                Strings.ProductTitle,
-                MessageBoxButtons.OK,
-                MessageBoxIcon.Warning);
+        public static void ShowPtvsdIncompatibleEnvError() {
+            ShowPtvsdMessage(
+                Strings.PtvsdIncompatibleEnvTitle,
+                Strings.PtvsdIncompatibleEnvMessage,
+                allowDisable: true,
+                isError: true
+            );
+        }
+
+        public static void ShowPtvsdModuleNotFoundError() {
+            ShowPtvsdMessage(
+                Strings.ImportPtvsdModuleNotFoundTitle,
+                Strings.ImportPtvsdModuleNotFoundMessage,
+                allowDisable: false,
+                isError: true
+            );
+        }
+
+        private static void ShowPtvsdMessage(string main, string content, bool allowDisable, bool isError) {
```

Comment:
I feel like we should have some exception handling around this function. Can't say specifically why, though.